test/functional/mempool_package_limits.py

```diff
@@ -177,6 +179,102 @@ def test_desc_count_limits(self):
+        parent_value = (first_coin_a["amount"] - DEFAULT_FEE) / 2 # Deduct reasonable fee and make 2 outputs
+        inputs = [{"txid": first_coin_a["txid"], "vout": 0}]
+        outputs = [{self.address : parent_value}, {ADDRESS_BCRT1_P2WSH_OP_TRUE : parent_value}]
+        rawtx = node.createrawtransaction(inputs, outputs)
+
+        parent_signed = node.signrawtransactionwithkey(hexstring=rawtx, privkeys=self.privkeys)
+        assert parent_signed["complete"]
+        parent_tx = tx_from_hex(parent_signed["hex"])
+        parent_txid = parent_tx.rehash()
+        node.sendrawtransaction(parent_signed["hex"])
+
+        # Chain M2a...M23a
+        spk = parent_tx.vout[0].scriptPubKey.hex()
+        value = parent_value
+        txid = parent_txid
+        for i in range(22): # M2a... M23a
+            (tx, txhex, value, spk) = make_chain(node, self.address, self.privkeys, txid, value, 0, spk)
+            txid = tx.rehash()
+            node.sendrawtransaction(txhex)
+
+        # P1
+        value_p1 = (parent_value - DEFAULT_FEE) / 2 # Deduct reasonable fee and make 2 outputs
+        inputs_p1 = [{"txid": parent_txid, "vout": 1}]
+        outputs_p1 = [{self.address : value_p1}, {ADDRESS_BCRT1_P2WSH_OP_TRUE : value_p1}]
+        rawtx_p1 = node.createrawtransaction(inputs_p1, outputs_p1)
+        tx_child_p1 = tx_from_hex(rawtx_p1)
+        tx_child_p1.wit.vtxinwit = [CTxInWitness()]
+        tx_child_p1.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
+        tx_child_p1_hex = tx_child_p1.serialize().hex()
+        txid_child_p1 = tx_child_p1.rehash()
+        package_hex.append(tx_child_p1_hex)
+        parents_tx.append(tx_child_p1)
+        values.append(value_p1)
+        scripts.append(tx_child_p1.vout[0].scriptPubKey.hex())
+
+        # P3
+        value_p3 = value_p1 - DEFAULT_FEE
+        rawtx_p3 = node.createrawtransaction([{"txid": txid_child_p1, "vout": 1}], {self.address : value_p3})
+        tx_child_p3 = tx_from_hex(rawtx_p3)
+        tx_child_p3.wit.vtxinwit = [CTxInWitness()]
+        tx_child_p3.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
+        tx_child_p3_hex = tx_child_p3.serialize().hex()
+        package_hex.append(tx_child_p3_hex)
+
+        # M1b
+        first_coin_b = self.coins.pop()
+        spk = None
+        txid = first_coin_b["txid"]
+        value = first_coin_b["amount"]
+        (tx, txhex, value, spk) = make_chain(node, self.address, self.privkeys, txid, value, 0, spk)
+        txid = tx.rehash()
+        node.sendrawtransaction(txhex)
+        parents_tx.append(tx)
+        values.append(value)
+        scripts.append(spk)
+
+        # P2
+        child_hex = create_child_with_parents(node, self.address, self.privkeys, parents_tx, values, scripts)
+        package_hex.append(child_hex)
+
+        assert_equal(24, node.getmempoolinfo()["size"])
+        assert_equal(3, len(package_hex))
+        testres = node.testmempoolaccept(rawtxs=package_hex)
```

Comment:
Did you consider the suggestion about asserting success for [P1, P3] but failure for [P1, P2, P3] ?

Reply:
I have removed M1b as it was creating more confusion and as jnewbery pointed out the test would still fail regardless of M1b being present there and it was not adding any edgecase that it not already present.